scalafix-cli/src/main/scala/scalafix/internal/v1/Args.scala

```diff
@@ -507,23 +506,25 @@ object Args {
   implicit val callbackEncoder: ConfEncoder[ScalafixMainCallback] =
     ConfEncoder.StringEncoder.contramap(_.toString)
   implicit val argsEncoder: ConfEncoder[Args] = generic.deriveEncoder
-  implicit val absolutePathPrint: TPrint[AbsolutePath] =
-    TPrint.make[AbsolutePath](_ => "<path>")
-  implicit val pathMatcherPrint: TPrint[PathMatcher] =
-    TPrint.make[PathMatcher](_ => "<glob>")
-  implicit val confPrint: TPrint[Conf] =
-    TPrint.make[Conf](implicit cfg => TPrint.implicitly[ScalafixConfig].render)
+  implicit val absolutePathPrint: _root_.pprint.TPrint[AbsolutePath] =
```

Comment:
I updated the code to compile, but I *believe* (no tests seem to prevent regressions of https://github.com/scalacenter/scalafix/commit/7811980d7804f5be12a60170d772e2242575d777) that all these `pprint.TPrint` implicit are now effectively useless since they are no longer picked up by https://github.com/scalameta/metaconfig/blob/3238d741cb12f972f35f7f55413ee8e7a720416b/metaconfig-core/shared/src/main/scala-2/metaconfig/internal/Macros.scala#L265 for rules built against newer versions, and were captured at compile time for rules built against old versions.

Since scala 2.11 does build against an old metaconfig where pprint is not shaded, I am still wondering what's best:
1. start consider dropping scala 2.11 support (mostly useful for https://github.com/scalacenter/scalafix/issues/998)
1. drop these implicit (the real impact on the output should be assessed)
1. freeze a scala-2.11 version of `Args.scala`

Reply:
> I updated the code to compile, but I _believe_ (no tests seem to prevent regressions of [7811980](https://github.com/scalacenter/scalafix/commit/7811980d7804f5be12a60170d772e2242575d777)) that all these `pprint.TPrint` instances are now effectively useless since they are no longer picked up.

Confirmed
```diff
--- <unnamed>
+++ <unnamed>
@@ -1,20 +1,20 @@
 $ sbt "cli/runMain scalafix.cli.Cli --help"
 ...
 [info] running (fork) scalafix.cli.Cli --help
-[info] Scalafix 0.9.34+19-147646c0-SNAPSHOT
+[info] Scalafix 0.9.34+10-055d1d6b-SNAPSHOT
 [info] Usage: scalafix [options] [<path> ...]
 [info] Scalafix is a refactoring and linting tool. Scalafix supports both syntactic and
 [info] semantic linter and rewrite rules. Syntactic rules can run on source code
 [info] without compilation. Semantic rules can run on source code that has been
 [info] compiled with the SemanticDB compiler plugin.
 [info] Common options:
-[info]   --rules | -r [String ...] (default: [])
+[info]   --rules | -r List[String] (default: [])
 [info]     Scalafix rules to run, for example ExplicitResultTypes. The syntax for rules is
 [info]     documented in
 [info]     https://scalacenter.github.io/scalafix/docs/users/configuration#rules
-[info]   --files | -f [<path> ...] (default: [])
+[info]   --files | -f List[AbsolutePath] (default: [])
 [info]     Files or directories (recursively visited) to fix.
-[info]   --config <path> (default: null)
+[info]   --config Option[AbsolutePath] (default: null)
 [info]     File path to a .scalafix.conf configuration file. Defaults to .scalafix.conf in
 [info]     the current working directory, if any.
 [info]   --check
@@ -25,7 +25,7 @@
 [info]   --diff
 [info]     If set, only apply scalafix to added and edited files in git diff against the
 [info]     master branch.
-[info]   --diff-base String (default: null)
+[info]   --diff-base Option[String] (default: null)
 [info]     If set, only apply scalafix to added and edited files in git diff against a
 [info]     provided branch, commit or tag.
 [info]   --syntactic
@@ -46,20 +46,20 @@
 [info]     files that should be fixed must be compiled with semanticdb-scalac.
 [info]     Dependencies are required by rules like ExplicitResultTypes, but the
 [info]     dependencies do not need to be compiled with semanticdb-scalac.
-[info]   --sourceroot <path> (default: null)
+[info]   --sourceroot Option[AbsolutePath] (default: null)
 [info]     Absolute path passed to semanticdb with -P:semanticdb:sourceroot:<path>.
 [info]     Relative filenames persisted in the Semantic DB are absolutized by the
 [info]     sourceroot. Defaults to current working directory if not provided.
-[info]   --semanticdb-targetroots [<path> ...] (default: [])
+[info]   --semanticdb-targetroots List[AbsolutePath] (default: [])
 [info]     Absolute paths passed to semanticdb with -P:semanticdb:targetroot:<path>. Used
 [info]     to locate semanticdb files. By default, Scalafix will try to locate
 [info]     semanticdb files in the classpath
 [info]   --auto-classpath
 [info]     If set, automatically infer the --classpath flag by scanning for directories
 [info]     with META-INF/semanticdb
-[info]   --auto-classpath-roots [<path> ...] (default: [])
+[info]   --auto-classpath-roots List[AbsolutePath] (default: [])
 [info]     Additional directories to scan for --auto-classpath
-[info]   --scalac-options [String ...] (default: [])
+[info]   --scalac-options List[String] (default: [])
 [info]     The scala compiler options used to compile this --classpath, for example
 [info]     -Ywarn-unused-import
 [info]   --scala-version ScalaVersion (default: "2.13.8")
@@ -82,7 +82,7 @@
 [info]     autoload -U _scalafix
 [info]     ```
 [info] Less common options:
-[info]   --exclude [<glob> ...] (default: [])
+[info]   --exclude List[PathMatcher] (default: [])
 [info]     Unix-style glob for files to exclude from fixing. The glob syntax is defined by
 [info]     `nio.FileSystem.getPathMatcher`.
 [info]   --tool-classpath URLClassLoader (default: "<classloader>")
@@ -94,15 +94,15 @@
 [info]     If set, throw exception in the end instead of System.exit
 [info]   --no-stale-semanticdb
 [info]     Don't error on stale semanticdb files.
-[info]   --settings ScalafixConfig (default: {})
+[info]   --settings Conf (default: {})
 [info]     Custom settings to override .scalafix.conf
-[info]   --out-from String (default: null)
+[info]   --out-from Option[String] (default: null)
 [info]     Write fixed output to custom location instead of in-place. Regex is passed as
 [info]     first argument to file.replaceAll(--out-from, --out-to), requires --out-to.
-[info]   --out-to String (default: null)
+[info]   --out-to Option[String] (default: null)
 [info]     Companion of --out-from, string that is passed as second argument to
 [info]     fileToFix.replaceAll(--out-from, --out-to)
 [info]   --auto-suppress-linter-errors
 [info]     Insert /* scalafix:ok */ suppressions instead of reporting linter errors.
-[info]   --cwd <path> (default: "/Users/anabrice/Projects/git/scalafix/scalafix-cli")
+[info]   --cwd AbsolutePath (default: "/Users/anabrice/Projects/git/scalafix/scalafix-cli")
 [info]     The current working directory
```

Reply:
I went for option (3)